zebra/zserv.c

```diff
@@ -1401,8 +1402,13 @@ static void zread_route_del(ZAPI_HANDLER_ARGS)
 	if (CHECK_FLAG(api.message, ZAPI_MESSAGE_SRCPFX))
 		src_p = &api.src_prefix;
 
+	if (api.vrf_id == VRF_DEFAULT && api.tableid != 0)
+		table_id = api.tableid;
+	else
+		table_id = zvrf->table_id;
+
```

Comment:
When support for policy based routing for vrf netns, this code will not be unified.
I know my comment comes too late, but if we want to make common code, we should have taken two different fields for table_id : table_id for VRF lite, and table_id for Policy Routing.

I create a ticket to handle this : https://github.com/FRRouting/frr/issues/1960
But this is not blocking for this pull request